LockSession/handler.ts

```diff
@@ -60,7 +61,8 @@ type LockSessionClient = Client<"ApiKeyAuth">;
 const canLock = (
   user: HslJwtPayloadExtended | ExchangeJwtPayloadExtended
 ): boolean =>
-  ExchangeJwtPayloadExtended.is(user) || gte(user.spid_level, SpidLevel.L2);
+  isExchangeToken(user) ||
+  (HslJwtPayloadExtended.is(user) && gte(user.spid_level, SpidLevel.L2));
```

Comment:
Add unit test that covers the new scenario.

Reply:
The scenario was already covered, I only changed the form of the control
![image](https://github.com/pagopa/io-web-profile-backend/assets/107944358/8cd6aabd-7622-4199-92c2-903ec6ef3990)

Reply:
Sorry, let me explain better. Among the existing unit tests, I cannot find any where we pass a `magic_link` token resulting in a `ResponseErrorForbiddenNotAuthorized`. Similarly, I cannot find any with an L1 token.
They are both important for preventing regressions.